src/ls/definition.cr

```diff
@@ -4,7 +4,7 @@ module Mint
     class Definition < LSP::RequestMessage
       property params : LSP::TextDocumentPositionParams
 
-      def execute(server) : LSP::LocationLink | LSP::Location?
+      def execute(server) : Array(LSP::LocationLink | LSP::Location) | LSP::LocationLink | LSP::Location | Nil
```

Comment:
I'd prefer a more unified return type `Array(LSP::LocationLink | LSP::Location)`

Reply:
I thought the above matched the [specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_typeDefinition), but it turned out it doesn’t!

The return type for “Go to definition” can be any of `Location | Location[] | LocationLink[] | null`, but a singular `LocationLink` is not allowed - so we are currently out of spec 😅

Simplifying it to `Array(LSP::LocationLink | LSP::Location)` might be a good option, will have to have a think about how best to refactor all the definition methods.

Reply:
> Simplifying it to `Array(LSP::LocationLink | LSP::Location)` might be a good option, will have to have a think about how best to refactor all the definition methods.

IMO, that can be in a separate PR.